R/plotCorrelations.R

```diff
@@ -0,0 +1,319 @@
+#' @param coExpFile file name for the functional correlation
+#' graph. If not specified (NULL - default) the graph will be shown on screen. Supported file formats - pdf and png
+#' @param semSimFile if specified - will dave the semantic similarity
+#' correlation graph to file. Supported formats - pdf and png
+#' @param  coExpVsSemFile if sepcified - will save the functional correlation
+#' vs semantic similarity correlation to file. Supported formats - see ggsave.
+#' @param  pShape shape of the points in the functional vs co-expression
+#' graph. Example values - 1,16,22,23 etc. See ggplot.
+#' @return (list) ggplot graph.
+#'
+#' @author \href{https://orcid.org/0000-0002-8778-6442}{Denitsa Vasileva} (aut)
+#' @references H. Wickham. ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York, 2016.
+#' @references \href{https://cran.r-project.org/web/packages/corrplot/vignettes/corrplot-intro.html}{corrplot package}
+#' @examples
+#' \dontrun{
+#' plotCorrelations("PHALY")
+#' ## plots pairwise co-expression correlation graph, semantic
+#' ## similarity correlation graph and functional vs co-expression correlation
+#' ## graphs and returns the latter as ggplot list
+#' }
+#' @examples
+#' \dontrun{
+#' plotCorrelations("PHALY", coExpFile = "somefile.pdf") ## save co-expression correlation graph
+#' ## correlation graph into file 'somefile.pdf', and plots similarity correlation
+#' ## graph and functional vs co-expression correlation graphs and returns the
+#' ## latter as ggplot list
+#' }
+#' @export
+plotCorrelations <- function(bioSys,
+                             coExpFile = NULL,
+                             semSimFile = NULL,
+                             coExpVsSemFile = NULL,
+                             pShape = 16) {
+  sdf <- computeCorrelations()
+  if (!bioSys %in% sdf$System) {
+    stop("Unknown system passed as parameter.")
+  }
+
+  # Convert the data frame to a symmetrix matrix (required by corrplot)
+  sysDF <- sdf[
+    sdf$System == bioSys,
+    c("gene1", "gene2", "correlation", "goCorrelation")
+  ]
+
+  geneCount <- length(unique(sysDF$gene1))
+  coM <- matrix(nrow = geneCount, ncol = geneCount)
+  goM <- matrix(nrow = geneCount, ncol = geneCount)
+  colnames(coM) <- unique(sysDF$gene1)
+  rownames(coM) <- unique(sysDF$gene2)
+  colnames(goM) <- unique(sysDF$gene1)
+  rownames(goM) <- unique(sysDF$gene2)
+  for (i in 1:nrow(sysDF)) {
+    coM[
+      as.character(sysDF$gene1[i]),
+      as.character(sysDF$gene2[i])
+    ] <- as.numeric(sysDF$correlation[i])
+    coM[
+      as.character(sysDF$gene2[i]),
+      as.character(sysDF$gene1[i])
+    ] <- as.numeric(sysDF$correlation[i])
+    goM[
+      as.character(sysDF$gene1[i]),
+      as.character(sysDF$gene2[i])
+    ] <- as.numeric(sysDF$goCorrelation[i])
+    goM[
+      as.character(sysDF$gene2[i]),
+      as.character(sysDF$gene1[i])
+    ] <- as.numeric(sysDF$goCorrelation[i])
+  }
+  # R requires Cairo package - in order to save the image to file
```

Comment:
Cairo is in `Imports:` - so the package won't install if Cairo can't be installed.